src/main/java/com/redhat/devtools/lsp4ij/LSPIJEditorUtils.java

```diff
@@ -0,0 +1,323 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij;
+
+import com.intellij.ide.highlighter.custom.SyntaxTable;
+import com.intellij.openapi.fileTypes.FileType;
+import com.intellij.openapi.fileTypes.PlainTextFileType;
+import com.intellij.openapi.fileTypes.impl.AbstractFileType;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.util.CachedValueProvider;
+import com.intellij.psi.util.CachedValuesManager;
+import com.intellij.util.containers.ContainerUtil;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.plugins.textmate.TextMateFileType;
+import org.jetbrains.plugins.textmate.psi.TextMateFile;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Provides utility methods for editor support behaviors.
+ */
+@ApiStatus.Internal
+public final class LSPIJEditorUtils {
```

Comment:
There was starting to be multiple, very closely related utility classes for editor support features like this including `LSPIJTextMateUtils`, `LSPCodeBlockUtils`, and what would have become `LSPIJAbstractFileTypeUtils`. I just consolidated them into a single `LSPIJEditorUtils` given their very close relationship with one another. For the most part, these are all just moves from `LSPIJTextMateUtils` and `LSPCodeBlockUtils` with some renames, e.g., `isCodeBlockStart/EndChar()` to `isOpen/CloseBraceCharacter()`, and some additions for plain text/abstract file type files.